Source/WebKit/WebProcess/UserContent/WebUserContentController.h

```diff
@@ -49,7 +49,7 @@ class WebCompiledContentRuleListData;
 class WebUserMessageHandlerDescriptorProxy;
 enum class InjectUserScriptImmediately : bool;
 
-class WebUserContentController final : public WebCore::UserContentProvider, private IPC::MessageReceiver {
+class WebUserContentController final : public WebCore::UserContentProvider, public IPC::MessageReceiver {
```

Comment:
Is this needed because WeakPtr needs access to elements of IPC::MessageReceiver?

Reply:
`IPC::MessageReceiver` is the one that subclasses `CanMakeWeakPtr`. So if we don't do public inheritance of `IPC::MessageReceiver`, we can't construct a `WeakPtr< WebUserContentController>`.